src/App.js

```diff
@@ -7,25 +7,39 @@ import {
 import Container from './components/app-container';
 
 export default class App extends React.Component {
-    constructor(props) {
-        super(props);
-    }
+  constructor(props) {
+      super(props);
+  }
+
+  /**
+   * Local Storage must be set up before rendering the app because some
+   * filtered study space information is distributed to various pages.
+   */
+  componentWillMount() {
+      // TODO: Figure out why these keys are not reset when deleted and the
+      //   React server is restarted
```

Comment:
this is because localstorage is a file saved to the user's machine. this will reset once the user deletes their localstorage / clear browser data etc.